frontend/packages/noobaa-storage-plugin/src/components/capacity-breakdown/capacity-breakdown-card.tsx

```diff
@@ -59,21 +61,56 @@ const getDisableableSelectOptions = (dropdownItems: DropdownItems) => {
   ));
 };
 
+const secretResource: FirehoseResource = {
+  isList: false,
+  kind: SecretModel.kind,
+  prop: 'secret',
+  namespace: CEPH_STORAGE_NAMESPACE,
+  name: 'rook-ceph-external-cluster-details',
+};
+
 const BreakdownCard: React.FC = () => {
   const [serviceType, setServiceType] = React.useState(ServiceType.MCG);
   const [metricType, setMetricType] = React.useState(
     CapacityBreakdown.defaultMetrics[ServiceType.MCG],
   );
   const [isOpenServiceSelect, setServiceSelect] = React.useState(false);
   const [isOpenBreakdownSelect, setBreakdownSelect] = React.useState(false);
-  const RGW = useFlag(RGW_FLAG);
+  const isRGWSupported = useFlag(RGW_FLAG);
+  const prevRGWVal = React.useRef(null);
+
+  React.useEffect(() => {
+    if (isRGWSupported !== prevRGWVal.current) {
+      prevRGWVal.current = isRGWSupported;
+      if (isRGWSupported) {
+        setServiceType(ServiceType.ALL);
+        setMetricType(CapacityBreakdown.defaultMetrics[ServiceType.ALL]);
+      }
+    }
+  }, [isRGWSupported]);
+
+  const [secretData, secretLoaded, secretLoadError] = useK8sWatchResource<K8sResourceKind>(
+    secretResource,
+  );
+  const rgwPrefix = React.useMemo(() => {
+    if (isRGWSupported && secretLoaded && !secretLoadError) {
+      try {
+        return JSON.parse(atob(secretData.data?.external_cluster_details)).find(
+          (item) => item.name === 'ceph-rgw',
+        ).data.poolPrefix;
+      } catch {
```

Comment:
Same block is used above , better to extract it in a function, to maintain consistency

Reply:
These two are used across two files. I don't think this logic is generic enough to land in the helpers file.

Reply:
To be specific here, this can be taken out in a standalone function, so when we have any changes in BE , it will be easier.
```
JSON.parse(atob(secretData.data?.external_cluster_details)).find(
          (item) => item.name === 'ceph-rgw',
        ).data.poolPrefix;
``